drivers: spi: Support spim23 and spim24 instances

nordic-segl opened the pull request:
Extend SPI driver with possibility to use
- spim23, spim24,
- spis23, spis24.

nordic-segl committed:
drivers: spi: Support spim23 and spim24 instances

Extend SPI driver with possibility to use
- spim23, spim24,
- spis23, spis24.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl commented:
FYI: @nordic-piks , @kl-cruz , @mstasiaknordic

anangl commented on soc/nordic/validate_base_addresses.c:
```diff
@@ -319,6 +321,8 @@ CHECK_DT_REG(uart00, NRF_UARTE00);
 CHECK_DT_REG(uart21, NRF_UARTE21);
 CHECK_DT_REG(uart22, NRF_UARTE22);
+CHECK_DT_REG(uart23, NRF_UARTE23);
+CHECK_DT_REG(uart24, NRF_UARTE24);
```
Unintended change, I guess.

anangl replied:
> Intended 😄

Well, the commit message does not mention anything about UARTs.

> UART23 ,24 was added in this PR: https://github.com/zephyrproject-rtos/zephyr/pull/90373/files
> 
> Not sure if that was "complete".

The above change does not complete it either. `NRFX_UARTE23/24` Kconfig options and their translations to nrfx configuration macros are still missing.

nordic-segl committed:
modules: hal_nordic: nrfx: Add mising UARTE23/24 Kconfigs

Add UARTE23 and UARTE24 missing Kconfig options and their
translation to NRFX configuration macros.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>
